Source/Core/DolphinQt/Config/Graphics/EnhancementsWidget.cpp

```diff
@@ -77,10 +81,22 @@ void EnhancementsWidget::CreateWidgets()
       std::max(Config::Get(Config::GFX_EFB_SCALE), Config::Get(Config::GFX_MAX_EFB_SCALE));
   for (int scale = static_cast<int>(resolution_options.size()); scale <= max_efb_scale; scale++)
   {
-    resolution_options.append(tr("%1x Native (%2x%3)")
-                                  .arg(QString::number(scale),
-                                       QString::number(static_cast<int>(EFB_WIDTH) * scale),
-                                       QString::number(static_cast<int>(EFB_HEIGHT) * scale)));
+    const QString scale_text = QString::number(scale);
+    const QString width_text = QString::number(static_cast<int>(EFB_WIDTH) * scale);
+    const QString height_text = QString::number(static_cast<int>(EFB_HEIGHT) * scale);
+    const int extra_index = resolution_options.size() - 2;
+    const QString& extra_text = resolution_extra_options.size() > extra_index ?
+                                    resolution_extra_options[extra_index] :
+                                    QStringLiteral("");
```

Comment:
Err I don't think it's technically legal to assign the temporary string to a reference here? The QStringLiteral() macro might do some magic that makes this work, I guess?

Reply:
Yeah I'm genuinely not sure why this works with QStringLiteral, because [this should be a compile error](https://godbolt.org/z/3jhddd4eP). For peace of mind please just put a QStringLiteral on the stack and reference that...

Reply:
I also thought it shouldn't have worked. I assumed that the literal string might have created some persistent static object. Or maybe it knows the reference will have the same lifetime of the stack variable so it could have a safe reference to it?

Are you saying it just compiles as a copy, as a compatibility fallback, because I'm referencing an r-value?

Reply:
I've made this a non reference variable even if it was likely fine as a reference